### ISSUE TITLE
feat request : allow hiding dimension in pivotted table

### ISSUE THREAD

**Issue by luutuankiet** [lightdash/lightdash#11448]
### Description
Hi, I have a use case in pivot table to enforce row order by a specific dimension - "hour" in the example.
In details my sort logic by dimensions are day > week > folder > then hour. This table is used to track habit over time.

Since the "hour" dimension is only for sorting, is it possible to hide this field while keeping its order logic intact?
![image](https://github.com/user-attachments/assets/e011f8cc-f8ac-4096-881e-4ba89803c379)

### What problem does this solve?
De clutter the viz. As seen from the demo, the hours column don't provide useful information thus might just confuse the consumers.

Thanks for looking this up. Let me know if you need additional info!

**Comment by ZeRego** [comment]
I don't think this is possible since all dimensions included are part of the group by which impacts metric calculation. 

For example, ff we hide the "is completed" dimension, would that display 2 rows for each year? Would we be ok with the user potentially creating a confusing table? 

<img width="1622" height="524" alt="Image" src="https://github.com/user-attachments/assets/4e80c070-f353-4e13-8c6e-6ff4dde9f700" />

Should we flip this feature request to be "Allow to sort by a dimension that is not part of the results"? Isn't this the real problem here that we are forced to add a column to the results just to sort by it?

**Comment by MattBreton** [comment]
There may be other reasons why people might want to hide rows. One very niche case would be filtering on the results of Window Functions. Window Functions go in the Table Calculations and for them to function correctly the constituent fields which make up that function need to be selected too.

As an example, take the case of filtering down on the latest date where records exist. You can create a Table Calculation which is the effect of `${date} = MAX(${date}) OVER()` then use the result of that in the filter pane. This only functions if `date` is also a selected column so now we're left with a `date` field which we would like to hide.

Sorting Values is definitely the main problem that hiding fields would solve, but there are other fringe cases too.

**Comment by torimwhaley** [comment]
Additional customer feedback:

**Use case:** A user has a display dimension (line-of-business name) and a companion sort dimension (integer 1-7) that enforces a specific business-defined sort order. They want to pivot on the display dimension with columns appearing in the order dictated by the sort dimension. Including the sort dimension in the pivot as a row or secondary pivot column causes the sort-key values to leak into the UI. `visible: false` on the sort dimension is ignored in pivoted tables.

**Current workaround:** Encoding the sort order into the display value itself (e.g., "1. NurseStat", "2. MedStat"), which pollutes every other chart, filter, and export that uses that dimension.

Custom business sort orders (not alphabetical, not numerical) are extremely common for product categories, organizational hierarchies, workflow stages, and regional priorities. An alternative solution could be a dimension property like `order_by: another_dimension_id` so a user-facing dimension can be sorted by a companion field without that companion showing up.


Related to this ticket: https://github.com/lightdash/lightdash/issues/9377
